core/src/test/java/hudson/FunctionsTest.java

```diff
@@ -510,24 +512,45 @@ public void printStackTrace(PrintWriter s) {
                 s.println("Some custom exception");
             }
         }, "Some custom exception\n", "Some custom exception\n");
-        /* TODO exact format changed in 11.0.9 (JDK-8226809 / JDK-8252444)
         // Circular references:
         Stack stack1 = new Stack("p.Exc1", "p.C.method1:17");
         Stack stack2 = new Stack("p.Exc2", "p.C.method2:27");
         stack1.cause(stack2);
         stack2.cause(stack1);
-        assertPrintThrowable(stack1,
-            "p.Exc1\n" +
-            "\tat p.C.method1(C.java:17)\n" +
-            "Caused by: p.Exc2\n" +
-            "\tat p.C.method2(C.java:27)\n" +
-            "\t[CIRCULAR REFERENCE:p.Exc1]\n",
-            "<cycle to p.Exc1>\n" +
-            "Caused: p.Exc2\n" +
-            "\tat p.C.method2(C.java:27)\n" +
-            "Caused: p.Exc1\n" +
-            "\tat p.C.method1(C.java:17)\n");
-        */
+        //Format changed in 11.0.9 / 8.0.272 (JDK-8226809 / JDK-8252444 / JDK-8252489)
+        if ((getVersion().getDigitAt(0) == 11 && getVersion().isNewerThanOrEqualTo(new VersionNumber("11.0.9"))) ||
+                (getVersion().getDigitAt(0) == 8 && getVersion().isNewerThanOrEqualTo(new VersionNumber("8.0.272")))) {
```

Comment:
Seems pretty fragile. Maybe better to just alter the `assertPrintThrowable` utility to tolerate either format.